src/todt.c

```diff
@@ -896,7 +896,7 @@ dt_t **toDtElem(TypeSArray *tsa, dt_t **pdt, Expression *e)
         pdt = dtend(pdt);
         Type *tnext = tsa->next;
         Type *tbn = tnext->toBasetype();
-        while (tbn->ty == Tsarray && (!e || tbn != e->type->nextOf()))
+        while (tbn->ty == Tsarray && (!e || !tbn->equivalent(e->type->nextOf())))
```

Comment:
The existence of `e->type->nextOf()` is guaranteed by CTFE interpretation of dataseg variable initializer in 2.070. I guess it had not been guaranteed yet in 2.068. 
So, if you want to backport this patch to 2.068, you would need to add `e->type->nextOf()` null check AND adjust the `len` calculation.